sdk/identity/azure-identity/azure/identity/_internal/msal_client.py

```diff
@@ -131,7 +131,8 @@ def _store_auth_error(self, response: PipelineResponse) -> None:
             # response to an auth request, so no credential will want to include it with an exception
             content = response.context.get(ContentDecodePolicy.CONTEXT_NAME)
             if content and "error" in content:
-                self._local.error = (content["error"], response.http_response)
+                error_code = content["error"] if isinstance(content, dict) else "oauth_error"
```

Comment:
In which case msal returns a string?

Isn't it a bug in msal?

Reply:
At this point, where we are storing the last error locally, we are reading the PipelineResponse from our pipeline (which MSAL calls since we pass MSALClient as the `http_client`), so this is outside of MSAL. 

A customer using a proxy in their environment was running into issues causing `content` to be a string instead of the assumed dictionary. This is an uncommon edge case, and this fix will allow the error http_response to propagated instead of erroring out with `string indices must be integers, not 'str'`.